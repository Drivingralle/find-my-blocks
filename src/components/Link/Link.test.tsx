import React from "react";
import renderer from "react-test-renderer";
import { Link } from "./";

it("render correctly", () => {
  const tree = renderer
    .create(<Link url="https://eddysims.com">Link</Link>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("can open in a new tab", () => {
  const tree = renderer
    .create(
      <Link url="https://eddysims.com" openInNewTab>
        Link
      </Link>
    )
    .toJSON();
  if (tree) {
    expect(tree.props.target).toBe("_blank");
    expect(tree).toMatchSnapshot();
  }
});

it("will render an icon", () => {
  const tree = renderer
    .create(
      <Link url="https://eddysims.com" openInNewTab icon="camera">
        Link
      </Link>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
