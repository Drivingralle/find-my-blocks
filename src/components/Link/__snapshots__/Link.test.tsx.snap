// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can open in a new tab 1`] = `
<a
  className="link"
  href="https://eddysims.com"
  target="_blank"
>
  Link
</a>
`;

exports[`render correctly 1`] = `
<a
  className="link"
  href="https://eddysims.com"
  target="_self"
>
  Link
</a>
`;

exports[`will render an icon 1`] = `
<a
  className="link"
  href="https://eddysims.com"
  target="_blank"
>
  <svg
    className="feather feather-camera "
    fill="none"
    height={24}
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
    viewBox="0 0 24 24"
    width={24}
  >
    <g
      dangerouslySetInnerHTML={
        Object {
          "__html": "<path d=\\"M23 19a2 2 0 0 1-2 2H3a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h4l2-3h6l2 3h4a2 2 0 0 1 2 2z\\"></path><circle cx=\\"12\\" cy=\\"13\\" r=\\"4\\"></circle>",
        }
      }
    />
  </svg>
  Link
</a>
`;
